### DISCUSSION TITLE
A user has multiple roles, just select one at login

### DISCUSSION THREAD

**Discussion by krloz1003** [spatie/laravel-permission#1794] [Show and tell]
Hi,
I have users with more than one role, in the morning they are in an office and in the afternoon they move them.

How can I make when logging in they select the role with which they are going to work and only the permissions of that role remain active?

I was thinking of adding **deleted_at** to the table of **model_has_role** and using it as _active_ or _inactive_.

Greetings

**Comment by drbyte** [comment]
Try this:
- use the Role from this package only to control permissions
- at login, let them choose which operating role to apply, and set that into a Session variable
- when checking authorization for the role, instead of only checking one thing (whether the user has the role), compare two things: does the user have the role, and does the session value match the role.

**Comment by erikn69** [comment]
Maybe similiar case https://github.com/spatie/laravel-permission/discussions/1782
This works for me
Add a extra field on table `roles` 
Overwrite `roles` relation on model `Users.php`
```
public function roles(): BelongsToMany
{
        return $this->morphToMany(
            config('permission.models.role'),
            'model',
            config('permission.table_names.model_has_roles'),
            config('permission.column_names.model_morph_key'),
            'role_id'
        )->where(time is morning or afternoon condition field);
}
```
`time is morning or afternoon condition field` you can validate with time or with a flag on login using `session`
